script/translations/develop.py

```diff
@@ -69,7 +69,7 @@ def substitute_translation_references(integration_strings, flattened_translation
 
 def substitute_reference(value, flattened_translations):
     """Substitute localization key references in a translation string."""
-    matches = re.findall(r"\[\%key:((?:[\w]+|[:]{2})*)\%\]", value)
+    matches = re.findall(r"\[\%key:((?:[\w-]+|[:]{2})*)\%\]", value)
```

Comment:
We could also use the same restrictions as `RE_TRANSLATION_KEY`: `[a-z0-9-_]`
https://github.com/home-assistant/core/blob/c88d4b09c9915348117a57c61bf85027fc7d939b/script/hassfest/translations.py#L24

```suggestion
    matches = re.findall(r"\[\%key:((?:[a-z0-9-_]+|[:]{2})*)\%\]", value)
```

Reply:
Yeah let's keep it the same 👍